hypothesis-python/src/hypothesis/provisional.py

```diff
@@ -119,12 +119,12 @@ def do_draw(self, data):
         # with a max of 255, that leaves 3 characters for a TLD.
         # Allowing any more subdomains would not leave enough
         # characters for even the shortest possible TLDs.
-        elements = cu.many(data, min_size=1, average_size=1, max_size=126)
+        elements = cu.many(data, min_size=1, average_size=3, max_size=126)
         while elements.more():
             # Generate a new valid subdomain using the regex strategy.
             sub_domain = data.draw(st.from_regex(self.label_regex, fullmatch=True))
             if len(domain) + len(sub_domain) >= self.max_length:
-                data.stop_example(discard=True)
+                elements.reject()
```

Comment:
For future reference, what was going wrong here?

Reply:
No semantic difference at all in this case, it's just a clearer idiom because the corresponding `start_example()` is an internal/implementation detail of `many()`.

Reply:
Ah, that makes sense.

Though looking at the underlying code, calling `many.reject()` followed by `break` might actually be introducing a bug, because `reject` doesn't actually close the example itself; it just tells the next `more` to close the example on its next loop.

But since we immediately break, that call to `more` never happens.